web/toolbar.js

```diff
@@ -174,6 +193,43 @@ class Toolbar {
       this.#adjustScaleWidth();
       this._updateUIState(true);
     });
+
+    this.#bindEditorToolsListener(options);
+  }
+
+  #bindEditorToolsListener({ editorFreeTextButton, editorSelectButton }) {
+    this.eventBus._on("annotationeditormodechanged", evt => {
+      const editorButtons = [
+        [AnnotationEditorType.FREE_TEXT, editorFreeTextButton],
+      ];
+
+      if (evt.enable === false) {
+        for (const [, button] of editorButtons) {
+          button.classList.remove("toggled");
+          button.setAttribute("aria-checked", false);
+          button.disabled = true;
+        }
+        editorSelectButton.classList.remove("toggled");
+        return;
+      } else if (evt.enable === true) {
+        for (const [, button] of editorButtons) {
+          button.disabled = false;
+        }
+        editorSelectButton.classList.add("toggled");
+      }
+
+      for (const [mode, button] of editorButtons) {
+        if (mode !== evt.mode) {
+          button.classList.remove("toggled");
+        } else if (button.classList.contains("toggled")) {
+          button.classList.remove("toggled");
+          button.setAttribute("aria-checked", false);
+        } else {
+          button.classList.add("toggled");
+          button.setAttribute("aria-checked", true);
+        }
+      }
```

Comment:
This feels overly complex, why not simply:
```suggestion
      const isNone = tool === AnnotationEditorType.NONE,
        isFreeText = tool === AnnotationEditorType.FREETEXT;

      editorNoneButton.classList.toggle("toggled", isNone);
      editorFreeTextButton.classList.toggle("toggled", isFreeText);

      editorNoneButton.setAttribute("aria-checked", isNone);
      editorFreeTextButton.setAttribute("aria-checked", isFreeText);
```